draft-dss-star.md

```diff
@@ -101,200 +101,567 @@ In this document we describe one such system, namely Distributed Secret Sharing
+  into `L` bytes of output keying material.
+- Nx: The output size of the `Extract()` function in bytes.
+
+This specification uses HKDF-SHA256 {{!HKDF=RFC5869}} as the KDF function, where Nx = 32.
+
+## Key-Committing Authenticated Encryption with Associated Data {#deps-aead}
+
+A Key-Committing Authenticated Encryption with Associated Data (KCAEAD) scheme is an algorithm
+for encrypting and authenticating plaintext with some additional auxiliary data.
+It has the following API and parameters:
+
+- `Seal(key, nonce, aad, pt)`: Encrypt and authenticate plaintext
+  `pt` with associated data `aad` using symmetric key `key` and nonce
+  `nonce`, yielding ciphertext and tag `ct`. This function
+    can raise a `MessageLimitReachedError` upon failure.
+- `Open(key, nonce, aad, ct)`: Decrypt ciphertext and tag `ct` using
+  associated data `aad` with symmetric key `key` and nonce `nonce`,
+  returning plaintext message `pt`. This function can raise an
+  `OpenError` or `MessageLimitReachedError` upon failure.
+- `Nk`: The length in bytes of a key for this algorithm.
+- `Nn`: The length in bytes of a nonce for this algorithm.
+- `Nt`: The length in bytes of the authentication tag for this algorithm.
+
+[[OPEN ISSUE: specify which KCAEAD to use]]
 
 # System Overview
 
-## Objective
+In STAR, clients generate encrypted measurements and send them to a single untrusted
+Aggregation Server in a report. Each report is effectively a random k-out-of-n share of
+the client data secret, along with some additional auxilary data. In a given amount of
+time, if the Aggregation Server receives the same encrypted value from k = REPORT_THRESHOLD
+clients, the server can recover the client data associated with each report. This ensures
+that clients only have their measurements revealed if they are part of a larger crowd,
+thereby achieving k-anonymity privacy (where k = REPORT_THRESHOLD).
 
-In STAR, clients generate `encrypted` measurements, that they send to a single untrusted Aggregation Server. In a given amount of time, if the Aggregation Server receives the same encrypted value from K clients (i.e. K values), the server is able to decrypt the value. This ensures that clients only have their measurements revealed if they are part of a larger crowd. This allows the client to maintain K-anonymity, when paired with mechanisms for removing client-identifying information from their requests.
+Each client report is as secret as the underlying underlying client data. That means low
+entropy client data values could be abused by an untrusted Aggregation Server in a
+dictionary attack to recover client data with fewer than REPORT_THRESHOLD honestly generated
+reports. To mitigate this, clients boost the entropy of their data using an Oblivious
+Pseudorandom Function (OPRF) provided by a separate, non-colluding Randomness Server.
 
-## System Architecture
+STAR also requires use of a client Anonymizing Server when interacting with the Aggregation
```

Comment:
```suggestion
STAR also requires use of a client Anonymizing Proxy when interacting with the Aggregation
```

Reply:
I see that DAP also uses this term: https://www.ietf.org/archive/id/draft-ietf-ppm-dap-01.html#name-anonymizing-proxies